app/src/main/java/ch/epfl/sweng/favors/EditProfileFragment.java

```diff
@@ -42,28 +44,31 @@ public void afterTextChanged(Editable editable) {
         }
     };
 
-
     private TextWatcherCustom profCityEditWatcher = new TextWatcherCustom() {
         @Override
         public void afterTextChanged(Editable editable) {
             User.getMain().set(User.StringFields.basedLocation, editable.toString());
         }
     };
 
-    private TextWatcherCustom profSexEditWatcher = new TextWatcherCustom() {
-        @Override
-        public void afterTextChanged(Editable editable) {
-            User.getMain().set(User.StringFields.sex, editable.toString());
+    private void displaySex(){
+        //Log.d(TAG, UserGender.getGenderFromUser(User.getMain()).toString());
+        UserGender gender = UserGender.getGenderFromUser(User.getMain());
+        Log.d(TAG,gender.toString());
+        switch (gender){
+            case F: binding.profSexEdit.check(R.id.profSexFEdit); break;
+            case M: binding.profSexEdit.check(R.id.profSexMEdit); break;
+            case DEFAULT: Log.e(TAG,"Gender parsing issue.");
         }
-    };
+    }
 
     @Override
-    public View onCreateView(LayoutInflater inflater, ViewGroup container,
-                             Bundle savedInstanceState) {
-
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
 
          binding = DataBindingUtil.inflate(inflater, R.layout.fragment_edit_profile,container,false);
          binding.setElements(this);
+         //Set the RadioGroup buttons to select the current sex
+         displaySex();
```

Comment:
This is setting the sex of the user to be prechecked depending on what he was currently selected